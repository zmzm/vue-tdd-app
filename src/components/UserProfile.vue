
<script>
export default {
  name: 'UserProfile'
}
</script>

<template>
  <div>
    UserProfile
  </div>
</template>