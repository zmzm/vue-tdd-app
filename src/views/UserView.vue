<script>
import UserSearchForm from '@/components/UserSearchForm';
import UserProfile from '@/components/UserProfile';

export default {
  name: 'UserView',
  components: {
    UserSearchForm,
    UserProfile,
  }
};
</script>

<template>
  <div>
    <UserSearchForm />
    <UserProfile />
  </div>
</template>