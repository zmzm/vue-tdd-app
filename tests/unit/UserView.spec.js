import { shallowMount } from '@vue/test-utils';

import UserView from '@/views/UserView';
import UserSearchForm from '@/components/UserSearchForm'
import UserProfile from '@/components/UserProfile'

describe('UserView', () => {
  it('renders the component', () => {
    // arrange
    const wrapper = shallowMount(UserView)

    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main child components', () => {
    // arrange
    const wrapper = shallowMount(UserView)
    const userSearchForm = wrapper.findComponent(UserSearchForm)
    const userProfile = wrapper.findComponent(UserProfile)

    // assert
    expect(userSearchForm.exists()).toBe(true)
    expect(userProfile.exists()).toBe(true)
  })
});
