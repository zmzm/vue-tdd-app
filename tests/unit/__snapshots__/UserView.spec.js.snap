// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders the component 1`] = `
<div>
  <user-search-form-stub></user-search-form-stub>
  <user-profile-stub></user-profile-stub>
</div>
`;
